Firestore/Source/Local/FSTMutationQueue.h

```diff
@@ -115,10 +116,21 @@ NS_ASSUME_NONNULL_BEGIN
  * don't contain the document key at all if it's convenient.
  */
 // TODO(mcg): This should really return an NSEnumerator
-// also for b/32992024, all backing stores should really index by document key
 - (NSArray<FSTMutationBatch *> *)allMutationBatchesAffectingDocumentKey:
     (const firebase::firestore::model::DocumentKey &)documentKey;
 
+/**
+ * Finds all mutation batches that could @em possibly affect the given document keys. Not all
+ * mutations in a batch will necessarily affect each key, so when looping through the batches you'll
+ * need to check that the mutation itself matches the key.
+ *
+ * Note that because of this requirement implementations are free to return mutation batches that
+ * don't contain any of the given document keys at all if it's convenient.
+ */
```

Comment:
Since the implementations return batches in sorted order, perhaps it's worth it to call it out in this method's description?

Reply:
I'm not sure that's a contractual thing we care about, and isn't something that can necessarily be guaranteed on the other platforms.

Reply:
@wilhuff Sorry to reopen old thread, but it has relevant context. @mikelehen Michael pointed out that it would matter for batches to be in order when [applying them](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/Source/Local/FSTLocalDocumentsView.mm#L74-L76). On Android, it's more natural to return batches unsorted, although of course it's possible to sort them as well.

Reply:
Ah right. Of course.

I think it's reasonable to add a comment that indicates these come out in sorted order. I don't think it's worth changing to a data structure that necessarily enforces that order though.

Reply:
Oh! I guess the implementation does actually do the right thing (I thought it didn't guarantee order) since std::set is apparently a sorted set.  So yeah, I guess we just need to update the comment.